deepchecks/base/dataset.py

```diff
@@ -424,32 +481,52 @@ def index_col(self) -> t.Optional[pd.Series]:
         """Return index column. Index can be a named column or DataFrame index.
 
         Returns:
-           If date column exists, returns a pandas Series of the index column.
-        """
-        if self._use_default_index is True:
-            return pd.Series(self.data.index)
+           If index column exists, returns a pandas Series of the index column.
+        """
+        if self._set_index_from_dataframe_index is True:
+            if self._index_name is None:
+                return pd.Series(self.data.index.get_level_values(0), name=self.data.index.name,
+                                 index=self.data.index)
+            elif isinstance(self._index_name, (str, int)):
+                return pd.Series(self.data.index.get_level_values(self._index_name), name=self.data.index.name,
+                                 index=self.data.index)
         elif self._index_name is not None:
             return self.data[self._index_name]
-        else:  # No meaningful index to use: Index column not configured, and use_column is False
+        else:  # No meaningful index to use: Index column not configured, and _set_index_from_dataframe_index is False
             return
 
     @property
-    def date_name(self) -> t.Optional[Hashable]:
-        """If date column exists, return its name.
+    def datetime_name(self) -> t.Optional[Hashable]:
+        """If datetime column exists, return its name.
 
         Returns:
-           (str) date column name
+           (str) datetime name
         """
-        return self._date_name
+        return self._datetime_name
+
+    def get_datetime_column_from_index(self, datetime_name):
```

Comment:
like yurii did in another PR, if this method doesn't use self, and called just once in init, than move it outside or make it classs method

Reply:
does use init

Reply:
English?